src/js/modal-dialog.js

```diff
@@ -0,0 +1,408 @@
+/**
+ * @file modal-dialog.js
+ */
+import document from 'global/document';
+
+import * as Dom from './utils/dom';
+import * as Events from './utils/events';
+import * as Fn from './utils/fn';
+import log from './utils/log';
+
+import Component from './component';
+import CloseButton from './close-button';
+
+const CLASS_NAME_PREFIX = 'vjs-modal-dialog';
+
+// An array of strings which are used for modal sub-element classes,
+// so will cause CSS issues if used as a slug.
+const DISALLOWED_SLUGS = [
+  'content'
+];
+
+const ESC = 27;
+
+/**
+ * The `ModalDialog` displays over the video and its controls, which blocks
+ * interaction with the player until it is closed.
+ *
+ * Modal dialogs include a "Close" button and will close when that button
+ * is activated - or when ESC is pressed anywhere.
+ *
+ * @extends Component
+ * @class ModalDialog
+ */
+class ModalDialog extends Component {
+
+  /**
+   * Constructor for modals.
+   *
+   * @param  {Player} player
+   * @param  {Object} [options]
+   * @param  {Mixed} [options.content=undefined]
```

Comment:
That's something we should verify. @heff @rcrooks Is there documentation about how the docs get built? [videojs/docs](/videojs/docs) is a bit sparse.